src/v/cloud_roles/aws_sts_refresh_impl.cc

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "cloud_roles/aws_sts_refresh_impl.h"
+
+#include "s3/client.h"
+#include "utils/file_io.h"
+
+#include <boost/property_tree/ptree.hpp>
+
+/// The path to credentials in the XML response returned by STS
+static constexpr std::string_view response_node_credential_path
+  = "AssumeRoleWithWebIdentityResponse.AssumeRoleWithWebIdentityResult."
+    "Credentials";
+
+/// These variables injected by the AWS operator must be present on the redpanda
+/// pod to ensure we can get credentials from STS.
+/// ref:
+/// https://docs.amazonaws.cn/en_us/eks/latest/userguide/specify-service-account-role.html
+struct aws_injected_env_vars {
+    static constexpr std::string_view role_arn = "AWS_ROLE_ARN";
+
+    static constexpr std::string_view token_file_path
+      = "AWS_WEB_IDENTITY_TOKEN_FILE";
+};
+
+struct sts_params {
+    static constexpr std::string_view role_session_name = "redpanda_si";
+
+    static constexpr std::string_view api_version = "2011-06-15";
+
+    static constexpr std::chrono::seconds token_expiry_seconds{3599};
+};
+
+struct sts_response_schema {
+    static constexpr std::string_view expiry = "Expiration";
+    static constexpr std::string_view access_key_id = "AccessKeyId";
+    static constexpr std::string_view secret_access_key = "SecretAccessKey";
+    static constexpr std::string_view session_token = "SessionToken";
+};
+
+static const boost::beast::string_view content_type{
+  "application/x-www-form-urlencoded"};
+
+static constexpr std::string_view request_payload
```

Comment:
I see this is meant for string interpolation but its easy to get something mixed up, maybe a helper can be used:
```
static ss::sstring build_request_payload(std::chrono::seconds s, int version, string session_name, ...){
  static constexpr std::string_view request_payload  
   = "Action=AssumeRoleWithWebIdentity"
    "&DurationSeconds={}"
    "&Version={}"
    "&RoleSessionName={}"
    "&RoleArn={}"
    "&WebIdentityToken={}";
    return fmt::format(request_payload, args......etc..);
}
```

Reply:
thanks, I have used named arguments now.